std/traits.d

```diff
@@ -5971,6 +5972,32 @@ enum bool isNarrowString(T) = (is(T : const char[]) || is(T : const wchar[])) &&
     }
 }
 
+/**
+ * Detects whether `T` is a comparable type. Basic types and structs and
+ * classes that implement opCmp are comparable.
+ */
+enum bool isComparable(T) = is(typeof(T.init < T.init));
```

Comment:
It's possible for T.init to be hijacked and make this test fail, e.g. `struct S { void init(){} }`, if someone has decided that init() is a nice name for their own initialization functions. A better way would be `is(typeof((T t) => t < t))`.